### PR TITLE
Missing downloads, sha256sum, broken URLs

### PR DESCRIPTION
**Pull request by erasche**

### CODE REVIEW

**Comment by bgruening** [comment]
Go for it @erasche looks good to me!

**Comment by erasche** [comment]
Sorry, should've been marked as WIP. We still need to check all the URLs, either @yhoogstrate has made progress on testing installation for every package, or I will tomorrow hopefully and

**Comment by yhoogstrate** [comment]
@erasche I can't figure out how to use planemo (shed_test?), to only check whether installation via tool_dependencies.xml fail or not.

**Comment by erasche** [comment]
Working on this now :)

tor. 30. jul. 2015 kl. 06.46 skrev yhoogstrate <notifications@github.com>:

**Comment by yhoogstrate** [comment]
@erasche Can we close this PR since you've been working on an alternative solution?

**Comment by erasche** [comment]
@yhoogstrate let's leave it open. The alternate solution will still need your sha256sums that you contributed in this PR, they'll just have different URLs.

I'll get the URLs updated in this PR soon.

**Comment by erasche** [comment]
@yhoogstrate finally made some more progress on this: I updated all URLs to CPC urls

Once I get an update pushed to the CPC (whenever jenkins is working again), I'll finish up this PR with the new clearer CPC urls and we can work on getting it merged. Sorry for the crazy delays.

**Comment by bgruening** [comment]
Not sure about the new URL's ... we should do a smarter job here and enhance the download_by_url param or add a new one that uses gsl.

**Comment by erasche** [comment]
@bgruening, I don't like them either, they're too opaque. As soon as jenkins is online, I'll be running the CPC upgrade which will rename packages to be much more transparent.

sure, someday we could have a sprecial CPC download action, but for now...I think URLs are best as no Galaxy changes required while the CPC is still "an experiment"

**Comment by bgruening** [comment]
@erasche I don't think we should rename the packages. The TS uncompress packages and changes into the directory with the same name of the package. If we change this we break recipes.

**Comment by erasche** [comment]
@bgruening it'll be fine, I promise :) The update I mention will revert archives to their old names so we don't break anything and I'll test this very extensively as I don't want to be the person who breaks everyone's packages. I'll also add the filename parameter that lets you specify what you should call the downloaded archive to further prevent issues. 

This is still WIP, let me finish this and then you can tell me all the things I'm doing wrong during review ;)

**Comment by bgruening** [comment]
@erasche this was not my intention :(

**Comment by erasche** [comment]
No, no, you're fine, I'm just teasing you a bit, it's all okay, really. :) Your comments are always helpful! Your reminded me of an edge case I'll want to test before moving out of WIP.

**Comment by erasche** [comment]
Okay. This is about ready to move out of WIP. I've been fighting with this PR for days and I'm about done with it. Really only the last thing to do is ensure that all of the DESeq packages have sha256sums. (not necessarily to CPC, but at least with hashes)

**Comment by yhoogstrate** [comment]
> This is about ready to move out of WIP.

Awesome!! Huge number of changes here, thanks @erasche 

> Really only the last thing to do is ensure that all of the DESeq packages have sha256sums. 

As far as I can see they're complete?

**Comment by erasche** [comment]
@yhoogstrate excellent, I was tired, I considered that I might have missed some. I'll get this rebased this evening.

**Comment by yhoogstrate** [comment]
Who can we sacrifice to manually test each package? :smiling_imp:

**Comment by erasche** [comment]
Yikes, yeah.

I'd like to see the ruby ones tested specifically since I had to change the `package==version` to a proper URL.

File: packages/package_fido_f63cd47b8414f87d57b8c1b713d34a04b29c7aee/tool_dependencies.xml [hunk 1]
```diff
@@ -1,7 +1,8 @@
+<?xml version="1.0"?>
 <tool_dependency>
     <package name="Fido" version="f63cd47b8414f87d57b8c1b713d34a04b29c7aee">
         <install version="1.0">
-            <actions>   
+            <actions>
                 <action type="shell_command">git clone --recursive https://github.com/hendrikweisser/Fido.git</action>
```
Comment (yhoogstrate) [inline packages/package_fido_f63cd47b8414f87d57b8c1b713d34a04b29c7aee/tool_dependencies.xml]: This repo contains no submodules so a recursive clone shouldn't be necessary. Maybe we can change it to using githubs revision specific `/archive/`-url like this: https://github.com/galaxyproject/tools-iuc/blob/master/test_repositories/golang_package/tool_dependencies.xml#L9

Comment (erasche) [inline packages/package_fido_f63cd47b8414f87d57b8c1b713d34a04b29c7aee/tool_dependencies.xml]: +1

**Comment by erasche** [comment]
Rebased, out of WIP :)

**Comment by martenson** [comment]
:clap:

File: packages/package_bedtools_2_22/tool_dependencies.xml [hunk 2]
```diff
@@ -3,7 +3,9 @@
     <package name="bedtools" version="2.22">
         <install version="1.0">
             <actions>
-                <action type="download_by_url">https://github.com/arq5x/bedtools2/releases/download/v2.22.1/bedtools-2.22.1.tar.gz</action>
+                <action type="download_by_url" sha256sum="43ab5da6b41350bbef78ea2ee5be0609fb02d76b920a3301b687a39af7017233">
+                    https://depot.galaxyproject.org/software/bedtools/bedtools_2.22_src_all.tar.gz
```
Comment (bgruening) [inline packages/package_bedtools_2_22/tool_dependencies.xml]: This downloads version 2.22 and not 2.22.1

Comment (erasche) [inline packages/package_bedtools_2_22/tool_dependencies.xml]: https://github.com/galaxyproject/cargo-port/blob/master/urls.tsv#L279

File: packages/package_matplotlib_1_2/tool_dependencies.xml [hunk 3]
```diff
@@ -21,9 +21,9 @@
     <package name="matplotlib" version="1.2.1">
         <install version="1.0">
             <actions>
-                <!-- first action is always downloading -->
-                <action type="download_by_url" target_filename="matplotlib-1.2.1.tar.gz">https://github.com/matplotlib/matplotlib/archive/v1.2.1.tar.gz#md5#47ee56f51200b8760b98d97ee86f57c6</action>
-
+                <action type="download_by_url" sha256sum="52e18972aed85f30b05cef41778ec77685df6012f0598cd216e996de9b9ea29b">
```
Comment (yhoogstrate) [inline packages/package_matplotlib_1_2/tool_dependencies.xml]: Should fix https://github.com/galaxyproject/tools-iuc/issues/519

Comment (erasche) [inline packages/package_matplotlib_1_2/tool_dependencies.xml]: Updated against master so this PR is merge-able

**Comment by martenson** [comment]
I am gonna merge this today unless someone screams `nooooooooooo` @galaxyproject/iuc

**Comment by martenson** [comment]
merged

**Comment by erasche** [comment]
Don't forget @yhoogstrate!

**Comment by martenson** [comment]
Youri thanks! @yhoogstrate

**Comment by martenson** [comment]
The changes from this PR did not make it to the MTS since it predates travis deployment. Any good way how to make travis update all of these? @erasche @yhoogstrate @bgruening @nsoranzo ?

**Comment by nsoranzo** [comment]
Packages are deprecated in tools-iuc, we can update a repo on the MTS when someone complains about it.

**Comment by martenson** [comment]
@nsoranzo good point, you are absolutely right
